tests/test_flux_xr.py

```diff
@@ -99,3 +66,21 @@ def test_transpose_invariance_noskin(self, chunks, skin_correction, order):
                 ii.transpose("dim_0", "dim_1", "dim_2"),
                 iii.transpose("dim_0", "dim_1", "dim_2"),
             )
+
+
+@pytest.mark.parametrize("skin_correction", [True, False])
+def test_all_input_array_sizes_valid(skin_correction):
+    shapes = (
+        (3, 4),
+        (2, 3, 4),
+        (2, 3, 4, 5),
+    )  # create_data() only allows for inputs of 2 or more dimensions
+    data = (create_data(s, skin_correction=skin_correction) for s in shapes)
+    if skin_correction:
+        func = skin
+    else:
+        func = noskin
+    tuple(func(*d, "coare3p0", 2, 10, 6) for d in data)
+    assert (
+        1 == 1
+    )  # This line is always true, but verifies that the above line doesn't crash the Fortran code
```

Comment:
```suggestion
@pytest.mark.parametrize('shape', [(3, 4), (2, 3, 4), (2, 3, 4, 5),])
def test_all_input_array_sizes_valid(skin_correction, shape):
    # create_data() only allows for inputs of 2 or more dimensions
    data = create_data(shape, skin_correction=skin_correction)
    if skin_correction:
        func = skin
    else:
        func = noskin
    func(*data, "coare3p0", 2, 10, 6)
```

Reply:
What I did here is factor out the shape as a parameterized input, so that each shape gets its own test. This enables a more fine grained control (e.g. if for some reason the 4d case fails, but the others pass we will see this immediately in the test report).